packages/core/core/src/plugin.ts

```diff
@@ -191,38 +169,12 @@ export class Plugin {
     return joinPath(this.pluginPath, filename);
   }
 
-  // TODO: add deploy provider
-  registerClientWeb3Provider(cb) {
-    this.clientWeb3Providers.push(cb);
-    this.addPluginType('clientWeb3Provider');
-  }
-
-  registerContractsGeneration(cb) {
-    this.contractsGenerators.push(cb);
-    this.addPluginType('contractGeneration');
-  }
-
-  registerCustomContractGenerator(cb) {
-    this.generateCustomContractCode = cb;
-    this.addPluginType('customContractGeneration');
-  }
-
-  registerTestContractFactory(cb) {
-    this.testContractFactory = cb;
-    this.addPluginType('testContractFactory');
-  }
```

Comment:
Those are used by RJ from Giveth iirc (at leats the last 2), do we remove them because they are "not used" or because we don't mind breaking?